### PR TITLE
8163229: several regression tests have a main method that is never executed

### PR DESCRIPTION
**Pull request by archiecobbs**
This PR deals with some housekeeping relating to compiler unit tests.

JDK-8163229 points out that there are several tests that have a `public static void main(String[])` method, but because the test expects to fail during compilation, the test is never actually run, and so these `main` methods are just clutter that can be removed.

However, there are also some tests where the test is never run, but the `main()` method generates one or more of the expected compilation errors. Obviously we need to keep those, but they don't need to be declared as `public static void main(String[])` which is misleading. So instead we rename these methods to `meth()`, and also we remove the `String[]` parameter when it's not needed by the test.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8163229](https://bugs.openjdk.org/browse/JDK-8163229): several regression tests have a main method that is never executed

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10992/head:pull/10992` \
`$ git checkout pull/10992`

Update a local copy of the PR: \
`$ git checkout pull/10992` \
`$ git pull https://git.openjdk.org/jdk pull/10992/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10992`

View PR using the GUI difftool: \
`$ git pr show -t 10992`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10992.diff">https://git.openjdk.org/jdk/pull/10992.diff</a>

</details>

### CODE REVIEW

File: test/langtools/tools/javac/4980495/static/Test.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 public class Test {
 
-    public static void main(String argv[]) {
+    public static void meth() {
```
Comment (vicente-romero-oracle) [inline test/langtools/tools/javac/4980495/static/Test.java:14]: not sure about this case but if for some tests the method doesn't need to be `static` we should remove the modifier

Comment (archiecobbs) [reply test/langtools/tools/javac/4980495/static/Test.java:14]: Yes - in fact, at first I started doing that as well, but it turns out that doing this wholesale generates a bunch of test failures, and therefore requires a good bit more analysis to figure out which tests can handle it and which can't. The primary goal here is to remove misleading `public static void main(String[])` methods, not minimize in general. Perhaps more a general test minimization effort could be relegated to "phase 2", i.e., another bug?

Comment (vicente-romero-oracle) [reply test/langtools/tools/javac/4980495/static/Test.java:14]: sure I'm OK with that

File: test/langtools/tools/javac/limits/ArrayDims2.java [hunk 2]
```diff
@@ -35,6 +35,4 @@ class ArrayDims2 {
         [][][][][][][][][][]
```
Comment (vicente-romero-oracle) [inline test/langtools/tools/javac/limits/ArrayDims2.java:35]: given that you are in the neighborhood :), the main method at: ArrayDims3.java can also be removed. Or can be done in another phase as you please

**Review by vicente-romero-oracle — APPROVED** [review]
just minor suggestions, for your consideration which can be addressed as part of a follow-up bug, lgtm

File: test/langtools/tools/javac/limits/ArrayDims2.java (hunk 2 above)
Comment (archiecobbs) [reply test/langtools/tools/javac/limits/ArrayDims2.java:35]: Sure, no problem... and same thing for `ArrayDims1.java` it looks like.

**Comment by vicente-romero-oracle** [comment]
I have updated the bug title to a more general one, now the PR title need to be synced with it.

**Review by vicente-romero-oracle — APPROVED** [review]
looks good thanks

**Comment by TheShermanTanker** [comment]
Verified, that the test failures are unrelated, will help you sponsor

**Pull request closed without merging**